### PR TITLE
Feature/richlist

### PR DESCRIPTION
**Pull request by Vritra4**
also includes:
- update formatter to `%+v` to make config.Print() more human friendly
- typo fix: BlockRecord member tag
- make config singleton
- update README.md
- 2 new envvars

and luna only for now.

### CODE REVIEW

File: indexer/height/client.go [hunk 1]
```diff
@@ -1,3 +1,15 @@
 package height
 
+import (
+	tmjson "github.com/tendermint/tendermint/libs/json"
+	tmdb "github.com/tendermint/tm-db"
+)
 
+func GetLastKownHeight(indexerDB tmdb.DB) (height HeightRecord, err error) {
```
Comment (jeffwoooo) [inline indexer/height/client.go:8]: Kown -> typo 
is this method called somewhere?

**Review by jeffwoooo — APPROVED** [review]
I wish indexer run after block batch flushing so we can read genesis state easily.

But that did break tx indexing accuracy when rollback, that's the reason we patched to put indexer before flush.

File: indexer/height/client.go (hunk 1 above)
Comment (Vritra4) [reply indexer/height/client.go:8]: no, it isn't.
it's a remaining from older implementation... i`ll remove it.

Comment (Vritra4) [reply indexer/height/client.go]: @jeffwoooo removed!

**Pull request merged**